app/models/discussion_reader.rb

```diff
@@ -97,12 +97,7 @@ def viewed!(age_of_last_read_item = nil)
       reset_counts!
     end
 
-    publish!
-  end
-
-  def publish!
-    publish_data = Simple::DiscussionSerializer.new(discussion, scope: { reader_cache: reader_cache })
-    MessageChannelService.publish(publish_data.as_json, to: user)
+    EventBus.broadcast('discussion_reader_viewed', discussion, self, user)
```

Comment:
I was trying to keep the broadcast super close to where the work is done.

Right now there is problem that when you create a discussion, other tabs don't see that it is read.

I don't think we're going to call DiscussionService#mark_as_read from DiscussionService#create are we?

Seems like the tidiest thing to do is fire an event whenever a discussion is viewed (ie: discussion_reader.viewed!).

There is heaps more I want to talk to you about today now I've had experience with this.. Would be great to pair again.